src/mcfacts/inputs/ReadInputs.py

```diff
@@ -228,6 +229,27 @@
     raise ValueError("[ReadInputs.py] Boolean data types are not allowed in"
                      "the INPUT_TYPES dictionary. Please use int instead.")
 
+
+def initialize_r_g(input_variables):
+    # pre-calculating r_g from the provided smbh_mass
+    # Initialize the shared value to None
+
+    c = ct.c.to('m/s')
+    G = ct.G.to('m^3/(kg s^2)')
+    # Assign units to smbh mass
+    if hasattr(input_variables["smbh_mass"], 'unit'):
+        smbh_mass = input_variables["smbh_mass"].to('solMass')
+    else:
+        smbh_mass = input_variables["smbh_mass"] * u.solMass
+    # convert smbh mass to kg
+    smbh_mass = smbh_mass.to('kg')
+    # Calculate r_g in SI
+    r_g_in_meters = G*smbh_mass/(c ** 2)
```

Comment:
Linting/style suggestion:
```suggestion
    r_g_in_meters = G * smbh_mass / (c ** 2)
```